button-inc/emissions-elt-demo
# bug: various data-sourced pages give 500 error in production

Addresses #233. Errors were occurring on various pages in the `prod` environment, these fixes address them so that the pages are reachable.
Additionally, adds a fix that was mentioned in [PR#234](https://github.com/button-inc/emissions-elt-demo/pull/235#discussion_r1187897791) for the breadcrumbs.

## Changes

- Adds further passing of secrets from GCP Secrets Manager into the Docker build.
- Removes console.logs that had been used for testing.
- Add config to API routes to enable resolution by external function (ie. next-auth).
- Addressed _static-to-dynamic at runtime_ error from Nextjs. See [these docs](https://nextjs.org/docs/messages/app-static-to-dynamic-error) for more information.

## To test

1. Navigate to the production page [http://34.125.36.177.nip.io/](http://34.125.36.177.nip.io/).
2. If you are logged in, log out. The list of authorization options should display on the home page. 
3. Log in, and navigate (click or type) to `/en/analyst/dataset/available`. The list of available datasets should display after a short loading time.
4. Navigate to `/en/analyst/imported`. The list of imported datasets should display. 
5. Navigate to `/en/analyst/anonymized`. The list of anonymized datasets should display.
